recipes/libcudss/meta.yaml

```diff
@@ -0,0 +1,157 @@
+
+{% set version = "0.3.0.9" %}
+{% set lib_version = version.split(".")[0]+"."+version.split(".")[1]+"."+version.split(".")[2] %}
+{% set platform = "linux-sbsa" %}  # [aarch64]
+{% set platform = "linux-x86_64" %}  # [linux64]
+{% set platform = "placeholder" %}  # [osx]
+{% set extension = "tar.xz" %}  # [unix]
+{% set platform = "windows-x86_64" %}  # [win64]
+{% set extension = "zip" %}  # [win64]
+
+{% set cuda_compiler_version = cuda_compiler_version|default("None") %}
+{% set cuda_major = environ.get("cuda_compiler_version", "12.0").split(".")[0] %}
+
+package:
+  name: libcudss-split
+  version: {{ version }}
+
+source:
+  url: https://developer.download.nvidia.com/compute/cudss/redist/libcudss/{{ platform }}/libcudss-{{ platform }}-{{ version }}_cuda{{ cuda_major }}-archive.{{ extension }}
+  sha256: 6eadd70e6442b43194e697c13080dae3cf1771257a334e7d7b658825a3e73a3b  # [aarch64 and (cuda_compiler_version or "").startswith("12")]
+  sha256: acc90b40334cd2e4409ec9e4200c8276f7acd50d61b7349b4ffdaeda4b2bf515  # [linux64 and (cuda_compiler_version or "").startswith("12")]
+  sha256: ac0979610d3eda9e6ce803380587118c6ec72b7dc580e6e96392ac065010196f  # [win64   and (cuda_compiler_version or "").startswith("12")]
+
+build:
+  number: 0
+  skip: true  # [cuda_compiler_version in (None, "None", "11.8")]
+  skip: true  # [not (linux64 or aarch64 or win64)]
+  script:
+    - cp -rv include $PREFIX  # [linux]
+    - cp -rv lib $PREFIX  # [linux]
+    - cp -rv include %LIBRARY_INC%  # [win]
+    - cp -rv lib %LIBRARY_LIB%  # [win]
+    - cp -rv bin %LIBRARY_BIN%  # [win]
+
+outputs:
+
+  - name: libcudss
+    build:
+      run_exports:
+        # Breaking changes every version until 1.0
+        - {{ pin_subpackage("libcudss", max_pin="x.x.x") }}
+    files:
+      - lib/libcudss.so.*  # [linux]
+      - lib/libcudss_commlayer_nccl.so.*  # [linux]
+      - lib/libcudss_commlayer_openmpi.so.*  # [linux]
+      - lib/libcudss.so.*  # [linux]
+      - bin/cudss64_0.dll  # [win64]
+    requirements:
+      build:
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+        - {{ compiler('cuda') }}
+        - arm-variant * {{ arm_variant_type }}  # [aarch64]
+        - {{ stdlib("c") }}
+      host:
+        - cuda-version {{ cuda_compiler_version }}  # [cuda_compiler_version != "None"]
+        - nccl
+        - openmpi
```

Comment:
nccl and openmpi are only required if customers want to use MGMN mode of cudss. I don't like imposing this requirement on customers who will only use cudss on a single GPU (and never enable MGMN mode). Or WDYT, is having an extra dependency for them is fine? Like there is no nccl for Windows, will it be a problem for using conda package of cudss on Windows?

Reply:
Good catch! We need to guard `nccl` with a platform selector. We can move the files which link to openmpi to a separate output, so users don't get them by default. Probably we will try to have a nompi build variant in order to match the rest of the channel.

Reply:
@kvoronin, do downstream packages which link against the nccl or openmpi comm layers need to also link against the same nccl/openmpi directly (they are public linkages)? Or are these private linkages which are abstracted away by the cudss API?

Either way, I need to know what versions of openmpi and nccl you link against, so that the recipe can have minimum versions of nccl and openmpi.

Reply:
I was looking at the headers because of making the compile test and it seems that the API does not mention nccl and openmpi, so whether to use either of these libraries is decided at runtime?